Add projection & predicate in test ID for BigQueryStorage

moscowart opened the pull request.

moscowart commented on scio-test/src/test/scala/com/spotify/scio/testing/JobTestTest.scala:
```diff
@@ -98,6 +98,16 @@ object BigQueryJob {
   }
 }
 
+object BigQueryJobReadingTheSameTableWthDifferentPredicateAndProjections {
```
Ugly long name. Should probably be moved to a separate test suite

regadas replied:
Yeah can we come up with a shorter one?

moscowart replied:
I'll try :D

moscowart commented on scio-test/src/test/scala/com/spotify/scio/testing/JobTestTest.scala:
```diff
@@ -380,6 +390,16 @@ class JobTestTest extends PipelineSpec {
+  def testBigQueryJobReadingTheSameTableWthDifferentPredicateAndProjections: Unit =
+    JobTest[BigQueryJobReadingTheSameTableWthDifferentPredicateAndProjections.type]
+      .args("--input=table.in")
+      .input(BigQueryIO[TableRow]("table.in"), (1 to 3).map(newTableRow))
```
add another input

moscowart replied:
done

moscowart commented on scio-test/src/test/scala/com/spotify/scio/testing/JobTestTest.scala:
```diff
@@ -380,6 +390,16 @@ class JobTestTest extends PipelineSpec {
+  def testBigQueryJobReadingTheSameTableWthDifferentPredicateAndProjections: Unit =
+    JobTest[BigQueryJobReadingTheSameTableWthDifferentPredicateAndProjections.type]
+      .args("--input=table.in")
+      .input(BigQueryIO[TableRow]("table.in"), (1 to 3).map(newTableRow))
```
add constructor for storage with predicate and projections

regadas replied:
👍 Yup that will make it look like:

```scala
BigQueryIO[TableRow]("table.in", selectedFields, rowRestriction), (1 to 3).map(newTableRow)
```

moscowart replied:
done

regadas reviewed (commented):
Thanks for working on this!

regadas commented on scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryIO.scala:
```diff
@@ -358,11 +358,17 @@ object BigQueryTable {
 
-  override def testId: String = s"BigQueryIO(${table.spec})"
+  override def testId: String =
+    s"""
```
as we talked this can follow the same pattern as the others

```scala
s"""BigQueryIO(${table.spec}, List(${selectedFields.mkString(",")), $rowRestriction)"""
```

moscowart replied:
done

moscowart commented:
Now comes the rebase :)

moscowart commented:
@regadas I added the same behaviour for typed BQ storage API and fixed failing tests. I also removed read params which seemed useless anymore. My only concern is the tests that I added because they don't match how other tests are implemented. So I would need some guidance there.

regadas commented on scio-google-cloud-platform/src/test/scala/com/spotify/scio/bigquery/BigQueryIOTest.scala:
```diff
@@ -125,10 +125,64 @@ final class BigQueyIOTest extends ScioIOSpec {
 
+  it should "read the same input table with different predicate and projections using bigQueryStorage" in {
+
+    JobTest[BigQueryJobReadingTheSameTableWithDifferentPredicateAndProjections.type]
```
here

moscowart replied:
alright

regadas reviewed (commented):
@moscowart Some minor comments but it looks good!

moscowart commented on scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/BigQueryIO.scala:
```diff
@@ -83,8 +83,8 @@ private object Reads {
     table: Table,
-    selectedFields: List[String] = BigQueryStorage.ReadParam.DefaultSelectFields,
-    rowRestriction: Option[String] = BigQueryStorage.ReadParam.DefaultRowRestriction
+    selectedFields: List[String] = Nil,
```
gotcha, i'll add them back for both untyped and typed api then

moscowart commented:
@regadas fixed all the comments. Is it worth adding any docs for these changes, e.g. migration guide, examples?

regadas reviewed (commented):
@moscowart Looks good! Some docs and migration guide would be nice.

re: Migration Guide we already have some info there for v0.10.0.
